.github/workflows/analysis_tests.yml

```diff
@@ -0,0 +1,36 @@
+---
+name: Run Python Analysis Tests
```

Comment:
After coming back to the analysis code after a break, there are a couple of things I'd like to consider. One of them is just that some pieces of the code are difficult to understand when you aren't deeply involved in the file and data structure. They're therefore likely easy to miss something about, so I'd like to see if while I familiarize myself with the code I can't write some tests to help show what certain methods are doing.